File: AudioKit/Operations/Math/AKSum.m
//
//  AKSum.m
//  AudioKit
//
//  Created by Aurelius Prochazka on 6/9/12.
//  Copyright (c) 2015 Aurelius Prochazka. All rights reserved.
//

#import "AKSum.h"

@implementation AKSum

- (NSString *)stringForCSD
{
    NSMutableArray *parameterStrings = [NSMutableArray array];
    
    for (AKParameter *param in self.inputs) {
        [parameterStrings addObject:[NSString stringWithFormat:@"AKAudio(%@)", param.parameterString]];
    }
    NSString *inputsCombined = [parameterStrings componentsJoinedByString:@", "];
    
    return [NSString stringWithFormat:@"%@ sum %@",self, inputsCombined];
}
@end

File: AudioKit/Operations/Math/AKSum.swift
//
//  AKSum.swift
//  AudioKit
//
//  Created by Aurelius Prochazka on 9/14/15.
//  Copyright © 2015 AudioKit. All rights reserved.
//

import Foundation

/** Addition helper function */
func + (left: AKParameter, right: AKParameter) -> AKSum {
    return AKSum(input: left, plus: right)
}

/** Sum of two input signals
*/
@objc class AKSum : AKParameter {
    
    // MARK: - Properties
    
    /** Input to the mathematical function */
    private var first  = AKParameter()
    /** Input to the mathematical function */
    private var second = AKParameter()
    
    // MARK: - Initializers
    
    /** Instantiates the sum
    - parameter input: The first input.
    - parameter plus: The second input.
    */
    init(input firstInput: AKParameter, plus secondInput: AKParameter)
    {
        super.init()
        first = firstInput
        second = secondInput
        dependencies = [first, second]
    }
    
    /** Computation of the next value */
    override func compute() {
        leftOutput  = first.leftOutput  + second.leftOutput
        rightOutput = first.rightOutput + second.rightOutput
    }
    
}
